tcms/rpc/tests/test_auth.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init
+
+from xmlrpc.client import Fault as XmlRPCFault
+from django.contrib.sessions.models import Session
+from tcms.rpc.api.user import User
+from tcms.rpc.tests.utils import APITestCase
+from tcms.tests import UserFactory
+
+
+class TestAuthLogin(APITestCase):
+    """Test Auth.login method"""
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        #
+        self.rpc_client.Auth.logout()
+        self.tester = UserFactory()
+        self.password = "password"
+        self.tester.set_password(self.password)
+        self.tester.save()
+
+    def test_login_with_username(self):
+        self.rpc_client.Auth.logout()
+        sesson_id = self.rpc_client.Auth.login(self.tester.username, self.password)
+        session = Session.objects.get(session_key=sesson_id).get_decoded()
+
+        self.assertIsNotNone(sesson_id)
+        self.assertEqual(User.objects.get(pk=session.get("_auth_user_id")), self.tester)
+
+    def test_login_without_username(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Username and password is required"):
+            self.rpc_client.Auth.login("", self.password)
+
+    def test_login_without_password(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Username and password is required"):
+            self.rpc_client.Auth.login(self.tester.username, "")
+
+    def test_login_with_incorrect_username(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Wrong username or password"):
+            self.rpc_client.Auth.login("kiwi_tester", self.password)
+
+    def test_login_with_incorrect_password(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Wrong username or password"):
+            self.rpc_client.Auth.login(self.tester.username, "kiwi-password")
+
+
+class TestAuthLogout(APITestCase):
+    """Test Auth.logout method"""
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.rpc_client.Auth.logout()
+        self.tester = UserFactory()
+        self.password = "password"
+        self.tester.set_password(self.password)
+        self.tester.save()
+
+    def test_logout(self):
+        sesson_id = self.rpc_client.Auth.login(self.tester.username, self.password)
+        self.assertIsNotNone(sesson_id)
+        self.assertTrue(Session.objects.filter(session_key=sesson_id).exists())
+
+        self.rpc_client.Auth.logout()
+        self.assertFalse(Session.objects.filter(session_key=sesson_id).exists())
```

Comment:
Again scenario is trying to check Django internals. While that could serve as a proof that user is successfully logged out this is again Django's job to verify that. 

A more practical example would be to logout and then call any other method which requires a login. Any of the filter methods is a good candidate, also Bug.details and Markdown.render require ONLY the user to be logged-in, no permissions.